modules/core/02-client/keeper/client_test.go

```diff
@@ -18,43 +18,71 @@ import (
-			suite.consensusState,
+			func() {
+				clientState = ibctm.NewClientState(testChainID, ibctm.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, testClientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath)
+				consensusState = suite.consensusState
+			},
 			true,
 		},
+		{
+			"failure: 07-tendermint client status is not active",
+			func() {
+				clientState = ibctm.NewClientState(testChainID, ibctm.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, testClientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath)
+				tmcs, ok := clientState.(*ibctm.ClientState)
+				suite.Require().True(ok)
+				tmcs.FrozenHeight = clienttypes.NewHeight(0, 1)
+				consensusState = suite.consensusState
+			},
+			false,
+		},
 		{
 			"success: 06-solomachine client type supported",
-			solomachine.NewClientState(0, &solomachine.ConsensusState{PublicKey: suite.solomachine.ConsensusState().PublicKey, Diversifier: suite.solomachine.Diversifier, Timestamp: suite.solomachine.Time}),
-			&solomachine.ConsensusState{PublicKey: suite.solomachine.ConsensusState().PublicKey, Diversifier: suite.solomachine.Diversifier, Timestamp: suite.solomachine.Time},
+			func() {
+				clientState = solomachine.NewClientState(0, &solomachine.ConsensusState{PublicKey: suite.solomachine.ConsensusState().PublicKey, Diversifier: suite.solomachine.Diversifier, Timestamp: suite.solomachine.Time})
+				consensusState = &solomachine.ConsensusState{PublicKey: suite.solomachine.ConsensusState().PublicKey, Diversifier: suite.solomachine.Diversifier, Timestamp: suite.solomachine.Time}
+			},
 			true,
 		},
 		{
 			"failure: 09-localhost client type not supported",
-			localhost.NewClientState(clienttypes.GetSelfHeight(suite.ctx)),
-			nil,
+			func() {
+				clientState = localhost.NewClientState(clienttypes.GetSelfHeight(suite.ctx))
+				consensusState = nil
+			},
 			false,
 		},
 	}
 
-	for i, tc := range cases {
-		clientID, err := suite.keeper.CreateClient(suite.ctx, tc.clientState, tc.consensusState)
-		if tc.expPass {
-			suite.Require().NoError(err, "valid test case %d failed: %s", i, tc.msg)
-			suite.Require().NotNil(clientID, "valid test case %d failed: %s", i, tc.msg)
-			suite.Require().True(suite.keeper.ClientStore(suite.ctx, clientID).Has(host.ClientStateKey()))
-		} else {
-			suite.Require().Error(err, "invalid test case %d passed: %s", i, tc.msg)
-			suite.Require().Equal("", clientID, "invalid test case %d passed: %s", i, tc.msg)
-			suite.Require().False(suite.keeper.ClientStore(suite.ctx, clientID).Has(host.ClientStateKey()))
-		}
+	for i, tc := range testCases {
+		tc := tc
+
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+			tc.malleate()
+
+			clientID, err := suite.keeper.CreateClient(suite.ctx, clientState, consensusState)
```

Comment:
while we are at it could this be:
```go
suite.chainA.GetSimApp().GetIBCKeeper().ClientKeeper.CreateClient()
```
The other keeper is [legacy](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/keeper_test.go#L85)

Reply:
ahh yea, it requiers a bit of diffs:
```diff
diff --git a/modules/core/02-client/keeper/client_test.go b/modules/core/02-client/keeper/client_test.go
index 012efa620..848b48617 100644
--- a/modules/core/02-client/keeper/client_test.go
+++ b/modules/core/02-client/keeper/client_test.go
@@ -9,7 +9,6 @@ import (
 
        clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
        commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
-       host "github.com/cosmos/ibc-go/v7/modules/core/24-host"
        "github.com/cosmos/ibc-go/v7/modules/core/exported"
        solomachine "github.com/cosmos/ibc-go/v7/modules/light-clients/06-solomachine"
        ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
@@ -58,7 +57,7 @@ func (suite *KeeperTestSuite) TestCreateClient() {
                {
                        "failure: 09-localhost client type not supported",
                        func() {
-                               clientState = localhost.NewClientState(clienttypes.GetSelfHeight(suite.ctx))
+                               clientState = localhost.NewClientState(clienttypes.GetSelfHeight(suite.chainA.GetContext()))
                                consensusState = nil
                        },
                        false,
@@ -72,15 +71,21 @@ func (suite *KeeperTestSuite) TestCreateClient() {
                        suite.SetupTest() // reset
                        tc.malleate()
 
-                       clientID, err := suite.keeper.CreateClient(suite.ctx, clientState, consensusState)
+                       clientID, err := suite.chainA.GetSimApp().IBCKeeper.ClientKeeper.CreateClient(suite.chainA.GetContext(), clientState, consensusState)
+
+                       // assert correct behaviour based on expected error
+                       clientState, found := suite.chainA.GetSimApp().IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientID)
+
                        if tc.expPass {
                                suite.Require().NoError(err)
                                suite.Require().NotNil(clientID)
-                               suite.Require().True(suite.keeper.ClientStore(suite.ctx, clientID).Has(host.ClientStateKey()))
+                               suite.Require().True(found)
+                               suite.Require().NotEmpty(clientState)
                        } else {
                                suite.Require().Error(err)
                                suite.Require().Equal("", clientID)
-                               suite.Require().False(suite.keeper.ClientStore(suite.ctx, clientID).Has(host.ClientStateKey()))
+                               suite.Require().False(found)
+                               suite.Require().Empty(clientState)
                        }
                })
        }
```

The issue is that the tests were still using `suite.ctx` which doesn't correspond to the context `suite.chainA` uses. I'm going to open a separate issue to migrate this test to the new format. I don't think we need to do it in this pr

Reply:
created #3531